From ac5f0d705c9dbd3190712996d21c4fb78d19e243 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Ball=C3=B3=20Gy=C3=B6rgy?= <ballogyor@gmail.com>
Date: Wed, 23 Apr 2025 20:35:09 +0200
Subject: [PATCH] Present existing window on activation

This ensures that the window is raised to the front on second launch.

Also, check if main window is exist before dismiss the toast on quit.
---
 src/app.vala | 7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/src/app.vala b/src/app.vala
index ce8e295ca..077366413 100644
--- a/src/app.vala
+++ b/src/app.vala
@@ -155,8 +155,10 @@ public void call_when_ready (owned CallReadyFunc func) {
     public override void activate () {
         base.activate ();
 
-        if (main_window != null)
+        if (main_window != null) {
+            main_window.present();
             return;
+        }
 
         var window = add_new_window ();
         window.set_state (UIState.COLLECTION);
@@ -296,7 +298,8 @@ public override void shutdown () {
             this.hold ();
         }
 
-        main_window.dismiss_toast ();
+        if (main_window != null)
+            main_window.dismiss_toast ();
 
         async_launcher.await_all ();
         suspend_machines ();
-- 
GitLab

